### PR TITLE
ACM-1915 Validate vSphere folder and resource pool

### PR DESCRIPTION
**Pull request by KevinFCormier**
- Ensures folder and resource pool paths are absolute and begin with the correct datacenter and cluster references
- Swaps the order of cluster and datacenter in the form, since datacenter is a higher-level concept

### CODE REVIEW

File: frontend/public/locales/en/translation.json [hunk 1]
```diff
@@ -1103,9 +1103,9 @@
     "credentialsForm.vCenter.wizardTitle": "Enter the VMware credentials",
     "credentialsForm.vsphereDiskType.labelHelp": "The disk provisioning method. This value defaults to the vSphere default storage policy if not set.",
     "credentialsForm.vsphereDiskType.placeholder": "Select the vSphere disk type",
-    "credentialsForm.vsphereFolder.labelHelp": "The absolute path of the vSphere folder to use, for example: /DATACENTER/vm/FOLDER",
+    "credentialsForm.vsphereFolder.labelHelp": "The absolute path (case-sensitive) of the vSphere folder to use, for example: /DATACENTER/vm/FOLDER",
```
Comment (oafischer) [inline frontend/public/locales/en/translation.json:1106]: Change this to 2 sentences, so:

"The absolute path (case-sensitive) of the vSphere folder to use. For example: /DATACENTER/vm/FOLDER"

File: frontend/public/locales/en/translation.json [hunk 2]
```diff
@@ -1103,9 +1103,9 @@
...
     "credentialsForm.vsphereFolder.placeholder": "Enter the absolute path to your vSphere folder",
-    "credentialsForm.vsphereResourcePool.labelHelp": "The absolute path of the vSphere resource pool to use, for example: /DATACENTER/host/CLUSTER/Resources/RESOURCE",
+    "credentialsForm.vsphereResourcePool.labelHelp": "The absolute path (case-sensitive) of the vSphere resource pool to use, for example: /DATACENTER/host/CLUSTER/Resources/RESOURCE",
```
Comment (oafischer) [inline frontend/public/locales/en/translation.json:1108]: Same here.

**Review by oafischer — APPROVED** [review]
Recommend the same change for 2 different lines, otherwise looks good. I'll approve in advance. Thanks!

**Review by RobDolares — APPROVED** [review]
/lgtm

**Pull request merged**